packages/js-sdk/tests/sandbox/files/list.test.ts

```diff
@@ -1,23 +1,101 @@
-import { assert } from 'vitest'
+import { assert, onTestFinished } from 'vitest'
 
 import { sandboxTest } from '../../setup.js'
 
 sandboxTest('list directory', async ({ sandbox }) => {
-  const dirName = 'test_directory4'
+  const parentDirName = 'test_directory'
 
-  await sandbox.files.makeDir(dirName)
+  await sandbox.files.makeDir(parentDirName)
+  await sandbox.files.makeDir(`${parentDirName}/subdir1`)
+  await sandbox.files.makeDir(`${parentDirName}/subdir2`)
+  await sandbox.files.makeDir(`${parentDirName}/subdir1/subdir1_1`)
+  await sandbox.files.makeDir(`${parentDirName}/subdir1/subdir1_2`)
+  await sandbox.files.makeDir(`${parentDirName}/subdir2/subdir2_1`)
+  await sandbox.files.makeDir(`${parentDirName}/subdir2/subdir2_2`)
 
-  const files = await sandbox.files.list(dirName)
-  assert.equal(files.length, 0)
+  const testCases = [
+    {
+      name: 'explicit depth 0 (should default to 1)',
+      depth: 0,
+      expectedLen: 2,
+      expectedFiles: ['subdir1', 'subdir2'],
+    },
+    {
+      name: 'default depth (1)',
+      depth: undefined,
+      expectedLen: 2,
+      expectedFiles: ['subdir1', 'subdir2'],
+    },
+    {
+      name: 'explicit depth 1',
+      depth: 1,
+      expectedLen: 2,
+      expectedFiles: ['subdir1', 'subdir2'],
+    },
+    {
+      name: 'explicit depth 2',
+      depth: 2,
+      expectedLen: 6,
+      expectedFiles: [
+        'subdir1',
+        'subdir1_1',
+        'subdir1_2',
+        'subdir2',
+        'subdir2_1',
+        'subdir2_2',
+      ],
+    },
+    {
+      name: 'explicit depth 3 (should be the same as depth 2)',
+      depth: 3,
+      expectedLen: 6,
+      expectedFiles: [
+        'subdir1',
+        'subdir1_1',
+        'subdir1_2',
+        'subdir2',
+        'subdir2_1',
+        'subdir2_2',
+      ],
+    },
+    {
+      name: 'negative depth should error',
+      depth: -1,
+      expectedLen: 0,
+      expectedFiles: [],
+      expectError: 'invalid_argument',
+    },
+  ]
 
-  await sandbox.files.write('test_directory4/test_file', 'test')
+  for (const testCase of testCases) {
+    if (testCase.expectError) {
```

Comment:
We can keep the loop, but I find the expectError part very confusing now, can you just do it like in go?